blockmanager.go

```diff
@@ -579,8 +579,39 @@ func (b *blockManager) handleBlockMsg(bmsg *blockMsg) {
 		return
 	}
 
+	// Meta-data about the new block this peer is reporting. We use this
+	// below to update this peer's lastest block height and the heights of
+	// other peers based on their last announced block sha. This allows us
+	// to dynamically update the block heights of peers, avoiding stale heights
+	// when looking for a new sync peer. Upon acceptance of a block or
+	// recognition of an orphan, we also use this information to update
+	// the block heights over other peers who's invs may have been ignored
+	// if we are actively syncing while the chain is not yet current or
+	// who may have lost the lock announcment race.
+	var heightUpdate int32
+	var blkShaUpdate *wire.ShaHash
+
 	// Request the parents for the orphan block from the peer that sent it.
 	if isOrphan {
+		// We've just received an orphan block from a peer. In order
+		// to update the height of the peer, we try to extract the
+		// block height from the scriptSig of the coinbase transaction.
+		// Extraction is only attempted if the block's version is
+		// high enough (ver 2+).
+		if blockchain.ShouldHaveSerializedBlockHeight(&bmsg.block.MsgBlock().Header) {
```

Comment:
This is definitely going way past the 80 mark.  A few chars here and there I don't say too much about, but this one is pushing 100.